### PR TITLE
Subscription integration test sec patch

### PR DESCRIPTION
**Pull request by jesse-dds**
Fixes issues highlighted by SonarQube.

image

image

image

image

image

### CODE REVIEW

File: atat/domain/csp/cloud/azure_cloud_provider.py [hunk 1]
```diff
@@ -184,19 +186,18 @@ def __init__(self, config, azure_sdk_provider=None):
...
             "client_secret": self.secret_key,
-            "resource": f"https://{self.sdk.cloud.suffixes.keyvault_dns[1:]}",
+            "resource": resource,
         }
         token_response = self.sdk.requests.get(url, data=payload, timeout=30)
         token_response.raise_for_status()
         token = token_response.json().get("access_token")
         if token is None:
-            message = (
-                f"Failed to get token for resource '{resource}' in tenant '{tenant_id}'"
```
Comment (jesse-dds) [inline atat/domain/csp/cloud/azure_cloud_provider.py:198]: `resource` and `tenant_id` were undefined

File: atat/domain/csp/cloud/azure_cloud_provider.py [hunk 2]
```diff
@@ -1244,15 +1244,8 @@ def _assign_billing_owner_role(self, graph_token, billing_admin_role_id, user_id
...
         url = f"{self.graph_resource}/beta/roleManagement/directory/roleAssignments"
-
         result = self.sdk.requests.post(url, headers=auth_header, json=request_body)
-
-        error = result.json().get("error")
-        if error and "A conflicting object" in error.get("message"):
-            return True
-
         result.raise_for_status()
-        return True
```
Comment (jesse-dds) [inline atat/domain/csp/cloud/azure_cloud_provider.py:1255]: All execution paths of this method returned `True` and the `True` value was never used or checked against.

File: atat/domain/csp/cloud/azure_cloud_provider.py [hunk 3]
```diff
@@ -1404,12 +1397,6 @@ def create_user_role(self, payload: UserRoleCSPPayload):
                 f"Failed to create user role assignment: {response.json()}"
             )
 
-    def _extract_subscription_id(self, subscription_url):
-        sub_id_match = SUBSCRIPTION_ID_REGEX.match(subscription_url)
-
-        if sub_id_match:
-            return sub_id_match.group(1)
```
Comment (jesse-dds) [inline atat/domain/csp/cloud/azure_cloud_provider.py:1411]: `SUBSCRIPTION_ID_REGEX` was undefined and the method was not used.

**Comment by carleton-dds** [comment]
@jesse-dds Can you provide a little bit of context in the description for what the SonarCube Issue this addresses is? Can you link to something or just copy/paste a quick description

**Pull request merged**